### PR TITLE
WIP Considering broadcasted elements in binop backwards

### PR DESCRIPTION
**Pull request by coreylowman**
Related to #491

### CODE REVIEW

File: src/tensor_ops/utilities/cpu_kernels.rs [hunk 1]
```diff
@@ -74,22 +74,127 @@ impl<E: Dtype, Op: BinaryDerivative<E>> BinaryKernel<Op, E> for Cpu {
...
+                        let go = out_buf[i_out];
+                        tmp += op.dfdx(l, r) * go;
+                        grad_rhs_buf[i_rhs] += op.dfdy(l, r) * go;
+                    }
+                    *gl += tmp;
+                }
+            } else {
+                // rhs has less data - iterate rhs in logical order
+                let [_, mut lhs_idx, mut out_idx] =
+                    index_for_binop(rhs.shape, rhs.strides, lhs.strides);
```
Comment (coreylowman) [inline src/tensor_ops/utilities/cpu_kernels.rs:116]: For both branches of the if, it's also possible that we can iterate in logical order if lhs_idx/out_idx are contiguous. This happens when the dims that are broadcast are already rightmost, e.g. softmax

**Comment by nkoppel** [comment]
Note: related to #520

**Comment by coreylowman** [comment]
FYI I'm not currently working on this (focusing on other things). If someone else wants to work on this feel free!

**Comment by coreylowman** [comment]
This no longer seems to speed things up, closing for now

**Pull request closed without merging**